draft-ietf-taps-interface.md

```diff
@@ -335,14 +335,15 @@ Listener := Preconnection.Listen()
 
 Listener -> ConnectionReceived<Connection>
 
-// Only receive complete messages
+// Only receive complete messages in a Conn.Received handler
 Connection.Receive()
 
 Connection -> Received(messageDataRequest, messageContext)
 
+//---- Receive event handler begin ----
```

Comment:
Technically, isn't everything after this line part of the receive handler? That also includes stopping listening—or at least listen stop shouldn't be called until ConnectionReceived fires.

Reply:
I thought about this and wasn't sure; I guess yes, since this code assumes only a single connection, it's easiest to put it inside the receive handler. Maybe we should change the comment line to say:
// Stop listening for incoming Connections (this example supports only one Connection)

Reply:
Yes, I guess it's good to say explicitly that this example receives a single connection and then stops listening.

I think it'd make sense to adjust the text above the example ("This is an example of how an application might listen for incoming Connections […]" --> "This is an example of how an application might listen for an incoming Connection").

To make it clearer that the handler is a separate piece of code, we could put it into a separate code snippet, above which we say something like "This is an example of a receive handler which gets called when an incoming Connection is received, reads a Message, sends a response, and then stops listening for incoming Connections".

Reply:
I thought about this too, but on the positive side, the example as it stands also reflects the sequence in which things happen - even if the code being part of the main code isn't quite right. My suggestion: I fix it as @tfpauly suggested and also incorporate your updated comment, in this PR, and once this lands, you can fine-tune it further with your suggestion to split the code out and we can see what folks think?

Reply:
Fine with me :)